helix-core/src/main/java/org/apache/helix/model/ParticipantHistory.java

```diff
@@ -112,6 +129,22 @@ public long getLastOfflineTime() {
     return offlineTime;
   }
 
+  /**
+   * Get the time when this node last goes offline in history. If the node does not have offline
+   * history or contains invalid date as the last element, return -1.
+   *
+   * @return
+   */
+  public long getLastTimeInOfflineHistory() {
```

Comment:
What is difference this w/ getLastOfflineTime?

Reply:
`getLastOfflineTime` returns LAST_OFFLINE_TIME, which can be either a timestamp or a -1 for ONLINE. `getLastTimeInOfflineHistory` returns the last timestamp in OFFLINE (list field), which is always a timestamp.